src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -140,6 +137,14 @@ public AddressBook load() throws StorageOperationException {
         } catch (IllegalValueException ive) {
             throw new StorageOperationException("File contains illegal data values; data type constraints not met");
         }
+        finally{
+        	if (!path.toFile().exists()){
+        		final AddressBook empty = new AddressBook();
+                save(empty);
+                System.out.println("File has been recreated");
+                return empty;
+        	}
+        }
```

Comment:
- This finally block can handle the exception thrown in the previous catch block. But this is not a neat way to code. 

-A better way is to throw the exception once you recover in the save() and handle it in executeCommand in Main.java. This way where away the save() is called, if the file is deleted the exception will be handled smoothly.

- You can even create a new Exception type for this case i.e. StorageOperationFileDeletedException